docs/DeveloperGuide.md

```diff
@@ -154,6 +154,117 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Manual Command
+
+#### Description
+
+The `manual` command displays the format for a specified command and a short description for a particular command. Since the field is not compulsory during the execution of the `manual` command, the user's input is being parsed in `ManualCommandParser`. After which, a new `ManualCommand` object will be created, and is executed by the `LogicManager`.
+
+#### Implementation
+1. Upon receiving the user input, the `LogicManager` starts to parse the given input text using `AddressBookParser#parseCommand()`.
+2. The `AddressBookParser` invokes the respective `Parser` based on the first word of the input text.
+3. Since the first word in the user input matches the word "manual", `AddressBookParser#parseCommand(arguments)` will be called. In this case, the arguments refer to the remaining input text after the exclusion of the command word "manual".
+    <div markdown="span" class="alert alert-info">:information_source:
+    **Note:** A ParseException will be thrown if the argument is invalid.
+    </div>
+
+4. The supplied argument will be trimmed using `String#trim()`.
+5. The `ManualCommandParser` will create a new `ManualCommand` using the (`argument` in Step 3) and returned to `LogicManager`.
+6. The `LogicManager` will call `ManualCommand#execute(Model model)`. If the `argument` is invalid, a `CommandException` will be thrown.
+7. Lastly, the `ManualCommand` will create a new `CommandResult` which will be returned to `LogicManager`.
+
+The following sequence diagram shows how the manual command works:
+
+![ManualCommandSequenceDiagram](images/ManualCommandSequenceDiagram.png)
+
+### Mark Command
+
+#### Description
+
+The `mark` command marks a specific undone task as done for a particular student. Since all the fields are compulsory during the execution of the `mark` command, the user's input is being parsed in `AddressBookParser`. After which, a new `MarkCommand` object will be created, and is executed by the `LogicManager`.
+
+#### Implementation
+1. Upon receiving the user input, the `LogicManager` starts to parse the given input text using `AddressBookParser#parseCommand()`.
+2. The `AddressBookParser` invokes the respective `Parser` based on the first word of the input text.
+3. Since the first word in the user input matches the word "mark", `AddressBookParser#parseCommand(arguments)` will be called. In this case, the arguments refer to the remaining input text after the exclusion of the command word "mark".
```

Comment:
```suggestion
3. Since the first word in the user input matches the word "mark", `MarkCommandParser#parse(arguments)` will be called. In this case, the arguments refer to the remaining input text after the exclusion of the command word "mark".
```